std/range/primitives.d

```diff
@@ -1987,6 +1993,177 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
 }
 
 /**
+Pops the first element from $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+Params:
+    r = Forward range
+
+Returns:
+    Next element of the input range
+ */
+auto frontPop(Range)(ref Range r)
+if (isForwardRange!Range)
+in
+{
+    assert(!r.empty, "empty range passed to " ~ __PRETTY_FUNCTION__);
+}
+body
+{
+    auto e = r.save.front;
```

Comment:
This is just downright weird and merits a comment. Also, if the issue is ranges with a transient `front`, as I recall, the conclusion of previous discussions on the topic was that Phobos isn't going to support them for anything except simple input ranges (and even that was iffy). If they happen to work, great, but too many algorithms simply won't work with them, and having to call `save` just because you want to call `front` is horrible. And the rest of Phobos definitely doesn't do that.

So, if we're doing it, then there should be a comment explaining it, but I'd strongly vote against calling `save`. It makes it so that this function doesn't work with input ranges just because of concern over a particular range type that already causes havoc with most algorithms anyway.